x-pack/plugins/lens/public/visualizations/xy/visualization.tsx

```diff
@@ -932,6 +932,94 @@ export const getXyVisualization = ({
       );
     }
 
+    const shouldRotate = state?.layers.length ? isHorizontalChart(state.layers) : false;
+    const dataLayers = getDataLayers(state.layers);
+    const axisGroups = getAxesConfiguration(dataLayers, shouldRotate, frame.activeData);
+    const logAxisGroups = axisGroups.filter(
+      ({ groupId }) =>
+        (groupId === 'left' && state.yLeftScale === 'log') ||
+        (groupId === 'right' && state.yRightScale === 'log')
+    );
+
+    if (logAxisGroups.length > 0) {
+      logAxisGroups
+        .map((axis) => {
+          const mixedDomainSeries = axis.series.filter((series) => {
+            let hasNegValues = false;
+            let hasPosValues = false;
+            const arr = activeData?.[series.layer]?.rows ?? [];
+            for (let index = 0; index < arr.length; index++) {
+              const value = arr[index][series.accessor];
+
+              if (value < 0) {
+                hasNegValues = true;
+              } else {
+                hasPosValues = true;
+              }
+
+              if (hasNegValues && hasPosValues) {
+                return true;
+              }
+            }
+
+            return false;
+          });
+          return {
+            ...axis,
+            mixedDomainSeries,
+          };
+        })
+        .forEach((axisGroup) => {
+          if (axisGroup.mixedDomainSeries.length === 0) return;
+          const { groupId } = axisGroup;
+
+          warnings.push({
+            uniqueId: `mixedLogScale-${groupId}`,
+            severity: 'warning',
+            shortMessage: '',
+            longMessage: (
+              <FormattedMessage
+                id="xpack.lens.xyVisualization.mixedLogScaleWarning"
+                defaultMessage="The {axisName} axis is set to a logarithmic scale but {n} of the datasets contains positive and negative data."
```

Comment:
This warning message doesn't really tell me that it is a problem.

Maybe, "When the {axisName} axis is set to logarithmic scale the dataset should not contain positive and negative data."